cypress/utils/utils.ts

```diff
@@ -339,6 +349,60 @@ export function clearAllFilters(): void {
     cy.contains(button, "Clear all filters").click({ force: true });
 }
 
+export function filterIssueBy(filterType: filterIssue, filterValue: string | string[]): void {
+    let selector = "";
+    selectFilter(filterType);
+    const isApplicableFilter =
+        filterType === filterIssue.appName ||
+        filterType === filterIssue.category ||
+        filterType === filterIssue.source ||
+        filterType === filterIssue.target;
+
+    if (isApplicableFilter) {
+        if (Array.isArray(filterValue)) {
+            filterValue.forEach((current) => {
+                inputText(searchInput, current);
+                click(searchButton);
+            });
+        } else {
+            inputText(searchInput, filterValue);
+            click(searchButton);
+        }
+    } else {
+        if (filterType == filterIssue.bs) {
+            selector = bsFilterName;
+        } else if (filterType == filterIssue.tags) {
+            selector = tagFilterName;
+        }
+        click(selector);
+        if (Array.isArray(filterValue)) {
+            filterValue.forEach((name) => {
+                clickByText(span, name);
+            });
+        } else {
+            clickByText(span, filterValue);
+        }
+        click(selector);
+    }
+}
```

Comment:
**suggestion:** This function has too much cognitive complexity, it can be reduced by doing something like we did in https://github.com/konveyor/tackle-ui-tests/blob/main/cypress/utils/utils.ts#L354

Reply:
I didn't want to repeat same code several times. I will add comments to make it more readable.